lib/9pfs/9pfs.h

```diff
@@ -39,63 +39,150 @@
 
 #include <vfscore/prex.h>
 
-/*
- * Currently supports only the 9P2000.u variant of the protocol.
+/**
+ * enum uk_9pfs_proto - Protocol version;
+ * currently only the 9P2000.u variant of the protocol is supported
  */
 enum uk_9pfs_proto {
+	/*
+	 * Some extensions and minor adjustments are added to
+	 * the basic protocol for Unix systems
+	 */
 	UK_9P_PROTO_2000U,
+	/*
+	 * Some extensions and minor adjustments are added to
+	 * the basic protocol for Linux systems
+	 */
 	UK_9P_PROTO_2000L,
+	/*
+	 * The number of existent protocols;
+	 * in this case, two (UK_9P_PROTO_2000U and UK_9P_PROTO_2000L)
+	 */
 	UK_9P_PROTO_MAX
 };
 
+/**
+ * struct uk_9pfs_mount_data - An entry containing the necessary data
+ * for mounting the filesystem
+ */
 struct uk_9pfs_mount_data {
-	/* 9P device. */
+	/* 9P device */
 	struct uk_9pdev		*dev;
-	/* Wanted transport. */
+	/* Wanted transport */
 	struct uk_9pdev_trans	*trans;
-	/* Protocol version used. */
+	/* Protocol version used */
 	enum uk_9pfs_proto	proto;
-	/* Username to attempt to mount as on the remote server. */
+	/* User name to attempt to mount as on the remote server */
 	char			*uname;
-	/* File tree to access when offered multiple exported filesystems. */
+	/*
+	 * File tree to access when the server is
+	 * offering several exported file systems
+	 */
 	char			*aname;
 };
 
+/**
+ * struct uk_9pfs_mount_data - An entry containing the necessary data
+ * for the operations that can be executed on files
+ */
 struct uk_9pfs_file_data {
-	/* Fid associated with the 9pfs file. */
+	/* Fid associated with the 9pfs file */
 	struct uk_9pfid        *fid;
 	/*
 	 * Buffer for persisting results from a 9P read operation across
-	 * readdir() calls.
+	 * readdir() calls
 	 */
 	char                   *readdir_buf;
 	/*
 	 * Offset within the buffer where the stat of the next child can
```

Comment:
What do you mean by stat?

Reply:
From my understanding it refers to the data extracted using [this](https://github.com/unikraft/unikraft/blob/staging/lib/uk9p/include/uk/9preq.h#L419) function which is called [here](https://github.com/unikraft/unikraft/blob/staging/lib/9pfs/9pfs_vnops.c#L544).